### PR TITLE
Add Connect To Messaging 2.0 API document

### PR DESCRIPTION
**Pull request by mdorjtseden**
This will be an API document for our new service Connect To Messaging 2.0. 
Note: Please don't merge until I sent an email.

### CODE REVIEW

**Review by ppatel-lp — APPROVED** [review]

File: pages/Documents/MessagingChannels/ConnectToMessaging/api.md [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+---
+
+### Introduction
+
+Connect To Messaging (C2M) is a product offering from LivePerson allowing brands to offer consumers an option to deflect to messaging when they call into their IVR. C2M API serves as an intermediary between the brand’s IVR System and LivePerson Conversational Cloud, ensuring that the consumer is invited to join a conversation with an agent via eligible messaging channels. Once a consumer responds to a message from that channel, C2M ensures that the conversation is routed to an agent of the appropriate skill specified by the brands.
+
+
+### Getting Started
+
+1. Onboarding to C2M is mandatory before it.
```
Comment (dliu-liveperson) [inline pages/Documents/MessagingChannels/ConnectToMessaging/api.md:20]: do we have a link for the onboarding doc?

File: pages/Documents/MessagingChannels/ConnectToMessaging/api.md [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+
+
+### API Specifications
+## API Domain
+
+C2M is deployed in three regions. **North America**, **EMEA**(Europe, Middle East and Africa), **APAC**(Asia Pacific). Use the domain api to identify the zone of C2M api which is to be used for an account.
+
+### Eligibility API
+
+Click [**Eligibility**](https://app.swaggerhub.com/apis-docs/nsavla/Connect_To_Messaging/1.0.0#/default/post_account__accountId__eligibility) to go through API spec and use example here to get started.
```
Comment (dliu-liveperson) [inline pages/Documents/MessagingChannels/ConnectToMessaging/api.md:33]: I do not think we should use a like like https://app.swaggerhub.com/apis-docs/**nsavla**/Connect_To_Messaging in the developer site.

File: pages/Documents/MessagingChannels/ConnectToMessaging/api.md [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+
+| Method | URI  |
+| :--- | :--- |
+| POST | https://{domain}/api/account/{ACCOUNT_ID}/eligibility?v={VERSION}
+
+**Path Parameters**
+
+| Name  | Description | Value/Example |
+| :--- | :--- | :--- |
+| domain   | see API Domain | va.connect-to-messaging.liveperson.net or lo.connect-to-messaging.liveperson.net or sy.connect-to-messaging.liveperson.net |
```
Comment (dliu-liveperson) [inline pages/Documents/MessagingChannels/ConnectToMessaging/api.md:43]: `see API Domain`, do you want to include the link here.

File: pages/Documents/MessagingChannels/ConnectToMessaging/api.md [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+
+**Request Body Parameters**
+
+| Name | Datatype | Required | Definition |
+| :--- | :--- | :--- |:--- |
+| skill | string | yes | Engagement skill |
+| consumerPhoneNumber | string | yes | Consumer’s phone number(E.164 format with leading "+") |
+| handoffId | string | yes | C2M handoff Id |
+| sde | array | no | Array of ctmrinfo and/or personal SDEs |
+| sdes.* | object | no | SDEs: ctmrinfo and/or personal. See from [here](https://developers.liveperson.com/engagement-attributes-types-of-engagement-attributes.html) |
```
Comment (dliu-liveperson) [inline pages/Documents/MessagingChannels/ConnectToMessaging/api.md:62]: I suggest to merge this row with the above row.

File: pages/Documents/MessagingChannels/ConnectToMessaging/api.md [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+    "recommendedChannelName": "sms",
+    "eligible": true,
+    "callId": "b52403dc-b140-45cc-a9ca-d749a39b1b56"
+}
+
+```
+
+### Invite API
+
+Click [**Invite**](https://app.swaggerhub.com/apis-docs/nsavla/Connect_To_Messaging/1.0.0#/default/post_account__accountId__invite) to go through API spec and use example here to get started.
```
Comment (dliu-liveperson) [inline pages/Documents/MessagingChannels/ConnectToMessaging/api.md:108]: ditto

File: pages/Documents/MessagingChannels/ConnectToMessaging/api.md [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+| Method | URI  |
+| :--- | :--- |
+| POST | https://{domain}/api/account/{ACCOUNT_ID}/eligibility?v={VERSION}
+
+**Path Parameters**
+
+| Name  | Description | Value/Example |
+| :--- | :--- | :--- |
+| domain   | see API Domain | va.connect-to-messaging.liveperson.net or lo.connect-to-messaging.liveperson.net or sy.connect-to-messaging.liveperson.net |
+| ACCOUNT_ID | LivePerson site ID | 34567231 |
```
Comment (dliu-liveperson) [inline pages/Documents/MessagingChannels/ConnectToMessaging/api.md:44]: make sure `34567231` is not a real account id

File: pages/Documents/MessagingChannels/ConnectToMessaging/api.md [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+{  
+  "errorMessage":"Not Found",
+  "errorCode":99
+}
+```
+
+| HTTP Status | Error Code | Error Message | Description/Resolution |
+| :--- | :--- | :--- |:--- |
+| 400 | 11 | Bad request | Onboarding is not completed |
+| 400 | 34 | | Onboarding is not completed |
```
Comment (dliu-liveperson) [inline pages/Documents/MessagingChannels/ConnectToMessaging/api.md:174]: onboarding is internal, so it should be removed

File: pages/Documents/MessagingChannels/ConnectToMessaging/api.md [hunk 8]
```diff
@@ -0,0 +1,254 @@
...
+| 500 | 100 | Internal Server Error | Database connection error |
+| 500 | 101 | Internal Server Error | Database query error |
+| 500 | 110 | Internal Server Error | Error fetching the handoff |
+| 500 | 130 | Internal Server Error | Error fetching the account setting |
+
+
+### Frequently Asked Questions
+
+<strong>What is the rate limit for the API?</strong>
+The current rate limit is 30 TPS/second/brand per API.
```
Comment (dliu-liveperson) [inline pages/Documents/MessagingChannels/ConnectToMessaging/api.md:208]: TPS is per second. so better to say 30 transactions per second per brand

File: pages/Documents/MessagingChannels/ConnectToMessaging/api.md [hunk 9]
```diff
@@ -0,0 +1,254 @@
...
+	```
+
+<strong>Do we need any other JWT other than APP JWT e.g. Consumer JWT?</strong>
+C2M service does not create or consume consumer JWT or other JWT except APP JWT. C2M API consumes AppJWT created from provided clientId and Secret for authentication.
+
+<strong>What should the authentication header look like, is the bearer token the only thing required even in production usage? Do we need to include our ConsumerKey/Secret or our AccessToken/Secret that we use in the 1.0 API at all, or any other information?</strong>
+App Jwt will be consumed as Bearer Token. No other key, secret or token will be consumed by C2M Messaging api.
+
+<strong>How does C2M 2.0 api provide status of message e.g. success/failure of delivery ?</strong>
+The C2M 2.0 campaign api is asynchronous meaning that the success and failure of a message to a recipient is noted only when the recipient is picked from the C2M internal queue and a message is sent as per pre configured message rate. The [conversations](#conversations-api-example-request-and-response) api will provide the status of recipients tied to the campaign created.
```
Comment (dliu-liveperson) [inline pages/Documents/MessagingChannels/ConnectToMessaging/api.md:235]: We will have the report API, link to that if the doc is live. If not, say stay tuned?

File: pages/Documents/MessagingChannels/ConnectToMessaging/api.md [hunk 10]
```diff
@@ -0,0 +1,254 @@
...
+
+<strong>Of the error cases described above, which of those errors should we consider "retry-able"? For example, a bad request due to a missing field is not retry-able because it will just always fail, but a case where one of the downstream services was temporarily unavailable could warrant a retry. Which error cases that we could get back from the /campaigns endpoint are retryable and how should we handle a retry to avoid sending duplicate messages to a customer?</strong>
+C2M Messaging service has retry mechanism internally on dependent services to reduce failures due to transient errors.
```
Comment (dliu-liveperson) [inline pages/Documents/MessagingChannels/ConnectToMessaging/api.md:238]: mention the rate limit 429 and recommended retry delay here?

File: pages/Documents/MessagingChannels/ConnectToMessaging/api.md [hunk 11]
```diff
@@ -0,0 +1,254 @@
...
+
+<strong>What’s the lookback period?</strong>
+- Lookback period is how long will LP services maintain context (like campaign info, skill etc) for a reply of a message that is sent to the recipient/consumer.  
+- Lookback period can be pre-configured up to 30 days. Current maximum lookback period is 30 days from when messages are sent using C2M API. Example: When a message is sent to consumer using C2M API and if consumer replies within 30 days from when message was sent, the response will be redirected to LE agent according to specified skill. A response after 30 days will not be treated as a conversation. Please note, if a consumer has an existing active conversation with a brand in any channel, the outbound message won’t be delivered.
+
+<strong>How do we know which field is optional or required?</strong>
+Refer to this document and [swagger](https://app.swaggerhub.com/apis-docs/nsavla/Connect_To_Messaging/1.0.0#/).
```
Comment (dliu-liveperson) [inline pages/Documents/MessagingChannels/ConnectToMessaging/api.md:245]: ditto

File: pages/Documents/MessagingChannels/ConnectToMessaging/api.md [hunk 12]
```diff
@@ -0,0 +1,254 @@
...
+
+
+| Field Name | Limitation |
```
Comment (dliu-liveperson) [inline pages/Documents/MessagingChannels/ConnectToMessaging/api.md:248]: what is this table for?

**Review by dliu-liveperson — CHANGES_REQUESTED** [review]

File: pages/Documents/MessagingChannels/ConnectToMessaging/api.md (hunk 3 above)
Comment (mdorjtseden) [reply pages/Documents/MessagingChannels/ConnectToMessaging/api.md:43]: This was discussed we will register this domain in csds first and provide the link.

File: pages/Documents/MessagingChannels/ConnectToMessaging/api.md (hunk 12 above)
Comment (mdorjtseden) [reply pages/Documents/MessagingChannels/ConnectToMessaging/api.md:248]: This one is the table header for the below.

File: pages/Documents/MessagingChannels/ConnectToMessaging/api.md (hunk 10 above)
Comment (mdorjtseden) [reply pages/Documents/MessagingChannels/ConnectToMessaging/api.md:238]: I'll put this explanation below the rate limit section.

File: pages/Documents/MessagingChannels/ConnectToMessaging/api.md (hunk 1 above)
Comment (mdorjtseden) [reply pages/Documents/MessagingChannels/ConnectToMessaging/api.md:20]: It's under review and will add the link then.

File: pages/Documents/MessagingChannels/ConnectToMessaging/api.md (hunk 12 above)
Comment (dliu-liveperson) [reply pages/Documents/MessagingChannels/ConnectToMessaging/api.md:250]: yes, but I do not see a line to talk about what the table is about. and I do not think it is related to the question above
`How do we know which field is optional or required`

Comment (mdorjtseden) [reply pages/Documents/MessagingChannels/ConnectToMessaging/api.md:250]: I'll add seperate question for this.

**Pull request merged**